src/main/java/com/mitchej123/hodgepodge/util/PathSanitizer.java

```diff
@@ -0,0 +1,26 @@
+package com.mitchej123.hodgepodge.util;
+
+import java.util.regex.Pattern;
+
+/**
+ * Utilities for sanitizing file paths generated with user input
+ */
+public final class PathSanitizer {
+
+    // This class just has some static utilities
+    private PathSanitizer() {}
+
+    // On Linux it's just /, macOS forbids / and :
+    // On Windows it's <>:"/\|?*
+    private static final Pattern ILLEGAL_FILE_NAME_CHARS = Pattern.compile("[/\\\\<>:\\\"|?*]");
```

Comment:
Wouldn't be `"[/\\<>:\"|?*]"` enough?
(https://regex101.com/r/5eAzR8/1)

Reply:
That is what the string is, just with backslashes&quotes escaped because of java's own string escape codes, isn't it?

Reply:
I think your IDE auto-escaped everything too. The string I posted is already escaped. Your string has an extra backslash for each backslash/quotation mark.

Reply:
<img width="398" alt="esc1" src="https://github.com/GTNewHorizons/Hodgepodge/assets/40035906/1a35aa09-2f4f-489d-a2cd-500991840d59">
<img width="369" alt="esc2" src="https://github.com/GTNewHorizons/Hodgepodge/assets/40035906/e1078b6a-394a-4786-a15c-70af137c9e1c">

Reply:
I tested it and the escape of the double-quote is not necessary, but it is for the backslash:
```java
private static final Pattern ILLEGAL_FILE_NAME_CHARS1 = Pattern.compile("[/\\\\<>:\\\"|?*]");
    private static final Pattern ILLEGAL_FILE_NAME_CHARS2 = Pattern.compile("[/\\<>:\"|?*]");
// escaping "a\\b/c\"d"
// 1: a-b-c-d
// 2: a\b-c-d
```